src/server/http/HTTPResponder.cpp

```diff
@@ -239,6 +285,19 @@ void HTTPResponder::generateResponse(HTTPClient &client) {
 		return;
 	}
 
+	// check autorization
+	if (route->getAuthBasic().size()) {
+		std::map<std::string, std::string>::iterator it = client.data.request.data.headers.find("AUTHORIZATION");
+		if (it == client.data.request.data.headers.end()) {
+			handleError(client, server, route, 401);
+			return ;
+		}
+		else if (!checkCredentials(route->getAuthBasicUserFile(), it->second)) {
```

Comment:
try catch this. depending on the error, throw 500 (syscalls break) or 400 (request invalid)